From 3f637e1918d542beb53beba5ee7fca1644a80f2d Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Fri, 19 Apr 2019 09:34:45 +0800
Subject: [PATCH 704/706] mscc: felix: Fix a false compiling warnings

drivers/net/ethernet/mscc/felix_board.c:286:18: warning: 'ts.tv_sec' may be used uninitialized in this function [-Wmaybe-uninitialized]
   tstamp_in_ns = ktime_set(ts.tv_sec, ts.tv_nsec);
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/net/ethernet/mscc/felix_board.c:286:41: warning: 'ts.tv_nsec' may be used uninitialized in this function [-Wmaybe-uninitialized]
   tstamp_in_ns = ktime_set(ts.tv_sec, ts.tv_nsec);
                                       ~~^~~~~~~~

"The warning is bogus and this case can not occur.  Apparently this is a false
positive resulting from gcc getting a little smarter about tracking assignments
but not smart enough." I agree with this.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/mscc/felix_board.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/mscc/felix_board.c b/drivers/net/ethernet/mscc/felix_board.c
index b4c45c20be7c..29047bf72cb8 100644
--- a/drivers/net/ethernet/mscc/felix_board.c
+++ b/drivers/net/ethernet/mscc/felix_board.c
@@ -235,7 +235,7 @@ static rx_handler_result_t felix_frm_ext_handler(struct sk_buff **pskb)
 	struct net_device *ndev = (*pskb)->dev;
 	struct sk_buff *skb = *pskb;
 	struct skb_shared_hwtstamps *shhwtstamps;
-	struct timespec64 ts;
+	struct timespec64 ts = { 0 };
 	struct ocelot_port *port = NULL;
 	struct ocelot *ocelot = NULL;
 	char *start = skb->data;
-- 
2.17.1

